marimo/_plugins/ui/_impl/anywidget/utils.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from marimo._dependencies.dependencies import DependencyManager
+from marimo._plugins.ui._impl.anywidget.types import (
+    BufferPaths,
+    TypedModelMessage,
+    WidgetModelState,
+    WidgetModelStateWithoutBuffers,
+)
+
+if TYPE_CHECKING:
+    from typing_extensions import TypeIs
+
+
+def extract_buffer_paths(
+    message: WidgetModelState,
+) -> tuple[WidgetModelStateWithoutBuffers, BufferPaths, list[bytes]]:
+    """
+    Extract buffer paths from a message.
+    """
+    DependencyManager.ipywidgets.require("for anywidget support.")
+    import ipywidgets  # type: ignore
+
+    _remove_buffers = ipywidgets.widgets.widget._remove_buffers  # type: ignore
+
+    # Get the initial state of the widget
+    state, buffer_paths, buffers = _remove_buffers(message)  # type: ignore
+
+    return state, buffer_paths, buffers  # type: ignore
```

Comment:
Not necessarily something to change in this PR, but in general I prefer avoiding blanket `# type: ignore` comments. If possible, I prefer ignoring specific error codes (e.g. `# type: ignore[attr-defined]`) so it’s clearer what mypy is skipping. However, as we've discussed, adding some type of lint rule to disallow this usage (across the codebase) would be the right approach and something outside the scope of this PR.